### ISSUE TITLE
Tileset heavily drops frames while panning or zooming if mouse cursor is over the active area.

### ISSUE THREAD

**Issue by viksl** [godotengine/godot#73360]
### Godot version

4.0 Beta RC2

### System information

Windows 10, Nvidia GTX 1660Ti, Vulkan

### Issue description

When you add atlas texture and make tiles from it and try to pan around the tileset or zoom in/out if your mouse cursor is over the active area (base tiles area) as in the video the FPS drop heavily. If mouse cursor never gets over the active area everything seems to be reasonably smooth.

https://youtu.be/ICz9zsGkg_w

### Steps to reproduce

Add TileMap
Add TileSet
Add Atlas
Add texture to the atlas
Pan around with mouse (middle mouse button) or try to zoom in/out when your cursor is over the base tiles area and when outside.

### Minimal reproduction project

[TilesetFPSdrop.zip](https://github.com/godotengine/godot/files/10743913/TilesetFPSdrop.zip)

**Comment by viksl** [comment]
Yeah I thought so too that the tileset might be pretty big for this. I don't think it's something that would needed to be in 4.0 just wanted to report it so it's here.

But, you say it draws tons of shapes (borders) and that causes the issue but that can't be it. When I zoom in/out or pan the borders remain the same so it's smooth when I zoom with cursor outside and when the cursor is inside but the displayed area is the same and the number of borderds/redraws too.

Could it be because when the mouse is over it has to draw the current active tile the mouse is over (white border) and there the draw call is triggered every frame on all of this?

I just noticed that even when you move mouse around the white border showing which tile us underneath the cursor is slow as hell too. Looks more like the redraw are inefficient maybe?

Again not a breaking change for 4.0.

**Comment by Loufe** [comment]
For what it's worth, I'm finding this happens quite a lot with our team. Would love to see it fixed.
